File: engine/engine-model/src/main/scala/tools/packet/buff/GivePirateBuff.scala
package tools.packet.buff

import client.MapleBuffStat
import net.opcodes.SendOpcode
import tools.Pair
import tools.packet.PacketInput

class GivePirateBuff(private var _statIncreases: java.util.List[Pair[MapleBuffStat, java.lang.Integer]],
                     private var _buffId: Int, private var _duration: Int) extends PacketInput {
  def statIncreases: java.util.List[Pair[MapleBuffStat, java.lang.Integer]] = _statIncreases

  def buffId: Int = _buffId

  def duration: Int = _duration

  override def opcode(): SendOpcode = SendOpcode.GIVE_BUFF
}

File: engine/engine-model/src/main/java/tools/packet/buff/GivePirateBuff.java
package tools.packet.buff;

import java.util.List;

import client.MapleBuffStat;
import net.opcodes.SendOpcode;
import tools.Pair;
import tools.packet.PacketInput;

public record GivePirateBuff(List<Pair<MapleBuffStat, Integer>> statIncreases, Integer buffId,
                             Integer duration) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GIVE_BUFF;
   }
}
